docs/advanced-guides/posts.md

```diff
@@ -364,16 +364,28 @@ The blob object, as JSON, would look something like:
 }
 ```
 
-The blob is then included in a `app.bsky.embed.images` array, along with an alt-text string. The `alt` field is required for each image. Pass an empty string if there is no alt text available.
+The blob is then included in a `app.bsky.embed.images` array, along with an alt-text string and the image dimensions.
+
+The `alt` field is required for each image. Pass an empty string if there is no alt text available.
+
+To get the image dimensions, you'll need a library like [Pillow](https://pillow.readthedocs.io/). Client apps need the image dimensions to display them without performance or visual issues.
 
 ```python
-post["embed"] = {
-    "$type": "app.bsky.embed.images",
-    "images": [{
-        "alt": IMAGE_ALT_TEXT,
-        "image": blob,
-    }],
-}
+from PIL import Image
+
+with Image.open(IMAGE_PATH) as im:
+    width, height = im.size
+    post["embed"] = {
```

Comment:
you can have just the `width, height = im.size` line in the `with` indent, and then move the `post["embed"] = ...` part back to left-justified. I think that reads a bit easier / is more idiomatic.

Reply:
to clarify:

```python
with Image.open(IMAGE_PATH) as im:
    width, height = im.size

post["embed"] = {
    "$type": "app.bsky.embed.images",
    "images": [{
        "alt": IMAGE_ALT_TEXT,
        "image": blob,
        "aspectRatio": {
            "width": width,
            "height": height
        }
    }],
}
```

Reply:
oh i guess there's no scopes haha